src/test/resources/all/maps/example2.vpr

```diff
@@ -0,0 +1,118 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+/**
+ * These examples and specifications are largely based on the Dafny examples at
+ * https://github.com/dafny-lang/dafny/blob/master/Test/dafny0/Maps.dfy,
+ * as on 11/11/2020; commit 'c477eecabd9948e63375e88fefcf2ad9ea62f69b'.
+ */
+ 
+method test1()
+{
+  var m : Map[Int, Int] := Map(2 := 3)
+  assert 2 in m
+  assert !(3 in m)
+  assert m[2] == 3
+  assert m[2 := 4] == Map(2 := 4)
+  assert m[7 := 1] == Map(2 := 3, 7 := 1)
+  assert m == Map(2 := 3)
+  
+}
+
+method test2()
+{
+  var m1 : Map[Int, Int] := Map(2 := 3)
+  var m2 : Map[Int, Int] := Map(3 := 2)
+  assert m1[m2[3]] == 3
+}
+
+method test3(m1 : Map[Int, Bool], m2 : Map[Int, Bool])
+  requires forall i : Int :: { i in m1 } { i in m2 } { m1[i] } { m2[i] }
+    0 <= i && i < 100 ==> i in m1 && i in m2 && m1[i] != m2[i]
+{
+  assert forall i : Int :: 0 <= i && i < 100 ==> m1[i] || m2[i]
+}
+
+method test4(m : Map[Int, Int])
+  requires forall i : Int :: 0 <= i && i < 100 ==> i in m && m[i] == i * i
+{
+  assert m[20] == 400
+}
+
+method test5()
+{
+  var m : Map[Int, Int] := Map(3 := 9)
+  
+  // this will verify in Carbon but will give a verification error in Silicon
```

Comment:
This needs an UnexpectedError annotation with the appropriate Silicon issue number